search/search-domain/src/main/java/io/camunda/search/filter/UsageMetricsFilter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.filter;
+
+import static io.camunda.util.CollectionUtil.addValuesToList;
+import static io.camunda.util.CollectionUtil.collectValues;
+
+import io.camunda.util.ObjectBuilder;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+public record UsageMetricsFilter(
+    List<String> events, Operation<OffsetDateTime> startTime, Operation<OffsetDateTime> endTime)
+    implements FilterBase {
+
+  public static final class Builder implements ObjectBuilder<UsageMetricsFilter> {
+    private List<String> events;
+    private Operation<OffsetDateTime> startTime;
```

Comment:
We can do that. However, the search API and the helper methods we have already allow to easily map date time operations, we can simply reuse this and thus allow this search to be flexible in this layer. The callers, e.g. the REST API, can decide how complex the query needs to be and what options are exposed.

Reply:
Looking into this a bit more closely, I agree with @megglos. I simplified this since the operation is hardwired by the query, this is not as flexible as other endpoints.